anonklub/anonklub
# feat(discord-bot): integrate with spartan-ecdsa

Fix #464 
Verify spartan ecdsa proofs with the discord bot.

I wished I had managed to use directly `@anonklub/spartan-ecdsa-wasm` directly in the discord bot server. But it is written as commonJS, which is not compatible with the [`--target bundler`](https://rustwasm.github.io/wasm-pack/book/commands/build.html#target) we used to packaged this wasm dep. So instead of converting the while discord bot code into ESM (to do later), I chose to add an api endpoint on the UI side (nextJS app). Then the discord bot is just calling that.

- [ ] Before merging, ensure new fly secrets are set (`VERIFICATION_CHANNEL_ID`, `VERIFIED_ROLE_ID`, `DISCORD_BOT_URL`, `UI_URL`)

## Test plan
0. Ensure you have the necessary env vars set
0. Create a server and add discord app/bot to it (use installation link provided in dev settings) 
1. Run discord bot: `pnpm --filter discord-bot start.dev`
2. Run ui: `pnpm --filter ui dev`
3. in `#verification` channel, use the `/verify` command
4. Follow instructions
